app/helpers/sessions_helper.rb

```diff
@@ -4,14 +4,35 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by(id: user_id) if session[:user_id]
+    elsif user_id = cookies.signed[:user_id]
+      user = User.find_by id: user_id
+      if user&.authenticated? :remember, cookies[:remember_token]
+        log_in user
+        @current_user = user
+      end
+    end
   end
 
   def logged_in?
     current_user.present?
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
permanent.signed để làm gì e ?

Reply:
Chưa trả lời này em

Reply:
- khi dùng cookies.permanent cookie này sẽ hết hạn sau một thời gian rất dài khoảng 20 năm từ thời điểm hiện tại
- signed : Nếu đặt id dưới dạng văn bản thuần túy sẽ làm lộ dạng cookie của ứng dụng và khiến kẻ tấn công dễ dàng xâm nhập tài khoản người dùng. Để tránh sự cố này, ta sử dụng cookie signed, cookie này sẽ mã hóa cookie một cách an toàn trước khi đặt nó trên trình duyệt